lib/ecto/changeset.ex

```diff
@@ -733,6 +743,17 @@ defmodule Ecto.Changeset do
       struct = model.__struct__
       query  = from m in struct, select: field(m, ^field), limit: 1
 
+      if opts[:scope] do
+        query = Enum.reduce(opts[:scope], query, fn(field, acc) ->
+          value = get_field(changeset, field)
+          if value do
```

Comment:
I don't think we should ignore the value. If it is nil, it also need to be used in our comparison, no?